internal/kafka/internal/services/quota/internal/utils/billing_model_resolver.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	amsv1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+)
+
+const (
+	CloudProviderAWS   = "aws"
+	CloudProviderRHM   = "rhm"
+	CloudProviderAzure = "azure"
+)
+
+var supportedCloudProviders = []string{CloudProviderAWS, CloudProviderRHM, CloudProviderAzure}
+
+/********************************************************************************************************************
+ * This is the root of the billing resolver chain.
+ * It demands the resolution of the billing model to the defined resolvers and stops the chain as soon as one of
+ * the resolvers is able to manage the request.
+ ********************************************************************************************************************/
+
+type BillingModelResolver interface {
+	SupportRequest(kafka *dbapi.KafkaRequest) bool
+	Resolve(orgID string, kafka *dbapi.KafkaRequest) (BillingModelDetails, error)
```

Comment:
What's the difference between the orgID parameter and the organization id field in the `kafka` parameter?
Is the first the AMS org id and the later the RH SSO external_id?